Lib/test/test_mimetypes.py

```diff
@@ -285,50 +286,53 @@ def test__all__(self):
-            mimetypes._main()
-            return output.getvalue().strip()
+    @classmethod
+    def to_string(cls, stream):
+        return stream.decode('ascii').strip()
 
-    def test_help_option(self):
-        support.patch(self, sys, "argv", [sys.executable, "-h"])
-        with support.captured_stdout() as output:
-            with self.assertRaises(SystemExit) as cm:
-                mimetypes._main()
+    @classmethod
+    def mimetypes_cmd(cls, *args, **kwargs):
+        result, _ = run_python_until_end('-m', 'mimetypes', *args)
+        return result.rc, cls.to_string(result.out), cls.to_string(result.err)
 
-        self.assertIn("Usage: mimetypes.py", output.getvalue())
-        self.assertEqual(cm.exception.code, 0)
+    def test_help_option(self):
+        retcode, out, err = self.mimetypes_cmd('-h')
+        self.assertEqual(retcode, 0)
+        self.assertIn('usage: mimetypes.py', out)
+        self.assertEqual(err, '')
 
     def test_invalid_option(self):
-        support.patch(self, sys, "argv", [sys.executable, "--invalid"])
-        with support.captured_stdout() as output:
-            with self.assertRaises(SystemExit) as cm:
-                mimetypes._main()
-
-        self.assertIn("Usage: mimetypes.py", output.getvalue())
-        self.assertEqual(cm.exception.code, 1)
+        retcode, out, err = self.mimetypes_cmd('--invalid')
+        self.assertEqual(retcode, 2)
+        self.assertEqual(out, '')
+        self.assertIn('usage: mimetypes.py', err)
 
     def test_guess_extension(self):
-        eq = self.assertEqual
+        retcode, out, err = self.mimetypes_cmd('-l', '-e', 'image/jpg')
+        self.assertEqual(retcode, 0)
+        self.assertEqual(out, '.jpg')
+        self.assertEqual(err, '')
 
-        extension = self.mimetypes_cmd("-l", "-e", "image/jpg")
-        eq(extension, ".jpg")
+        retcode, out, err = self.mimetypes_cmd('-e', 'image/jpg')
+        self.assertEqual(retcode, 0)
+        self.assertEqual(out, "I don't know anything about type image/jpg")
+        self.assertEqual(err, '')
 
-        extension = self.mimetypes_cmd("-e", "image/jpg")
-        eq(extension, "I don't know anything about type image/jpg")
-
-        extension = self.mimetypes_cmd("-e", "image/jpeg")
-        eq(extension, ".jpg")
+        retcode, out, err = self.mimetypes_cmd("-e", "image/jpeg")
+        self.assertEqual(retcode, 0)
+        self.assertEqual(out, '.jpg')
+        self.assertEqual(err, '')
 
     def test_guess_type(self):
-        eq = self.assertEqual
-
-        type_info = self.mimetypes_cmd("-l", "foo.pic")
-        eq(type_info, "type: image/pict encoding: None")
-
-        type_info = self.mimetypes_cmd("foo.pic")
-        eq(type_info, "I don't know anything about type foo.pic")
+        retcode, out, err = self.mimetypes_cmd('-l', 'foo.pic')
+        self.assertEqual(retcode, 0)
+        self.assertEqual(out, "type: image/pict encoding: None")
+        self.assertEqual(err, '')
```

Comment:
For some reason, macOS started to return `image/x-pict` for `.pic` failing an assertion and I cannot find out why:

```plain
======================================================================
FAIL: test_guess_type (test.test_mimetypes.MimetypesCliTestCase.test_guess_type)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/runner/work/cpython/cpython/Lib/test/test_mimetypes.py", line 329, in test_guess_type
    self.assertEqual(out, "type: image/pict encoding: None")
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: 'type: image/x-pict encoding: None' != 'type: image/pict encoding: None'
- type: image/x-pict encoding: None
?             --
+ type: image/pict encoding: None
```